.github/workflows/release.yaml

```diff
@@ -89,44 +89,71 @@ jobs:
         env:
           GOOS: ${{ matrix.platform.os }}
           GOARCH: ${{ matrix.platform.arch }}
-          VERSION: ${{ needs.set-version.outputs.version }}
+          VERSION: ${{ github.event.release.tag_name }}
         run: |
           # if pack is true, use build-and-pack target (which compresses the binary), otherwise use build target
           if [ "${{ matrix.platform.pack }}" = "true" ]; then
             make build-and-pack
           else
             make build
           fi
-      - name: Upload binary to GitHub Release
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          # Define binary path
-          BINARY_PATH="_output/${{ needs.set-version.outputs.version }}/istio-usage-collector-${{ matrix.platform.os }}-${{ matrix.platform.arch }}"
-          
-          # For Windows builds, the .exe extension is part of the filename
-          if [ "${{ matrix.platform.os }}" = "windows" ]; then
-            BINARY_PATH="${BINARY_PATH}.exe"
-          fi
-
-          # Verify binary exists
-          if [ ! -f "$BINARY_PATH" ]; then
-            echo "::error::Binary not found at $BINARY_PATH"
-            ls -la _output/${{ needs.set-version.outputs.version }}/
-            exit 1
-          fi
 
-          SHA256_PATH="${BINARY_PATH}.sha256"          
-          # Verify the checksum file exists
-          if [ ! -f "$SHA256_PATH" ]; then
-            echo "::error::Checksum file not found at $SHA256_PATH"
-            ls -la _output/${{ needs.set-version.outputs.version }}/
-            exit 1
-          fi
+  publish-binaries:
+    name: Publish Binaries
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    needs: [build-binaries]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Authenticate to Google Cloud Storage
+        uses: 'google-github-actions/auth@v2'
+        with:
+          credentials_json: ${{ secrets.GC_TODO_GHA_SA }}
+          project_id: solo-public
+      - name: Set up Google Cloud SDK
+        uses: 'google-github-actions/setup-gcloud@v2'
+      - name: Publish Binaries to Google Cloud Storage
+        uses: 'google-github-actions/upload-cloud-storage@v2'
+        with:
+          path: '_output' # Publish the version folder to the bucket
+          destination: istio-usage-collector
+          parent: false
+          process_gcloudignore: false
+  
+  publish-release-metadata:
```

Comment:
similar to another tool, i've added this to create some manifests/metadata(?) so that people can simply do a `curl <link> | sh` to download their relevant _latest_ version, instead of needing to manually go through the storage bucket. this is especially useful for docs purposes to provide a one-liner download script.